### PR TITLE
Improve contract - part2

### PR DESCRIPTION
**Pull request by JakeLin**
Address the comments in previous PR #1 . @xavierlepretre please have a look, thanks

### CODE REVIEW

File: migrations/2_deploy_contracts.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+var SafeMath = artifacts.require("./SafeMath.sol");
+var Splitter = artifacts.require("./Splitter.sol");
+
+var bob = "0x978ddD545eA2a1C73B690Fa8557Edf748e385252";
+var carol = "0x2c6A4E13E56c93B86aE9bf6030E6C1062f61ff62";
```
Comment (xavierlepretre) [inline migrations/2_deploy_contracts.js:5]: Don't hard-code. Remember that `module.exports = function(deployer, network, accounts) {`.

File: contracts/Splitter.sol [hunk 2]
```diff
@@ -23,25 +31,28 @@ contract Splitter {
...
 
   function withdraw() external {
     require(msg.sender == bob || msg.sender == carol, "Only Bob or Carol can withdraw!");
     if (msg.sender == bob) {
-      require(bobBalance > 0, "Bob's balance must be grater than zero!");
       uint256 tempBobBalance = bobBalance;
+      require(tempBobBalance > 0, "Bob's balance must be grater than zero!");
       bobBalance = 0;
       msg.sender.transfer(tempBobBalance);
+      emit LogBobWithdrawn(tempBobBalance);
```
Comment (xavierlepretre) [inline contracts/Splitter.sol:49]: If the recipient emits an event on reception, in which order will these 2 events be?

**Review by xavierlepretre — COMMENTED** [review]
Good.
Now that the contract is good with Bob and Carol, it is time to refactor and make it such that it can work with any 2 accounts as decided by the sender. Without restrictions.
Merge this and create another PR for clarity.

File: contracts/Splitter.sol (hunk 2 above)
Comment (JakeLin) [reply contracts/Splitter.sol:49]: The reception event will emit first since the transfer runs first, is it right?

Comment (xavierlepretre) [reply contracts/Splitter.sol:49]: Yep. Is it an order you like? `LogReceived`, then `LogWithdrawn`.

Comment (JakeLin) [reply contracts/Splitter.sol:49]: you are right, should be the other way around, thanks

**Pull request merged**